components/SearchBar/index.tsx

```diff
@@ -61,38 +73,55 @@ function Component() {
 
   const checkDomain = () => {
     const path = window.location.pathname.replace("/", "").toLowerCase();
-    const dom = path.split('.')[1];
-    if (
-      dom === 'tyron' || 'did' || 'vc' || 'treasury'
-    ) {
+    if (path.split('.')[1] === 'did' || path.split('.')[1] === 'vc' || path.split('.')[1] === 'treasury') {
       return true
     } else {
       return false
     }
   }
 
-  useEffect(() => {
-
-    //@todo review
+  const setUsername = () => {
     const path = window.location.pathname.replace("/", "").toLowerCase();
+    if (checkPath()) {
+      return path;
+    } else if (checkDomain()) {
+      return path.split('.')[0];
+    } else if (path.includes('.did') && path.includes('/')) {
+      return path.split('/')[0].split('.')[0]
+    } else if (path.split('/')[1] === 'did' || path.split('/')[1] === 'funds' || path.split('/')[1] === 'recovery') {
+      return path.split('/')[0]
+    } else {
+      return username
+    }
+  } 
 
-    //@todo reorg the following so it is easier to read
-    const name_ = checkPath() ? path : checkDomain() ? path.split('.')[0] : path.split('/')[0];
-    const domain_ = checkPath() ? 'did' : checkDomain() ? path.split('.')[1] : 'did';
-    setName(name_); setDomain(domain_);
-
-    if (path !== "") {
+  const setDomain = () => {
+    const path = window.location.pathname.replace("/", "").toLowerCase();
+    if (checkPath()) {
+      return 'did';
+    } else if (checkDomain()) {
+      return path.split('.')[1];
+    } else if (path.split('/')[1] === 'did' || path.split('/')[1] === 'funds' || path.split('/')[1] === 'recovery') {
+      return 'did';
+    } else {
+      return domain;
+    }
+  }
+  
+  useEffect(() => {
+    const path = window.location.pathname.replace("/", "").toLowerCase();
+    
+    if (path.split('/')[1] === 'xwallet' && !is_admin?.verified) {
```

Comment:
This should always be the case, not only when is_admin.verified.

Reply:
No, I already try it. If we remove is admin.verified we never can go to /xwallet

Reply:
Shouldn't it be when admin is not verified? If we redirect username/xwallet to /username, does this mean we don't need /xwallet anymore? How would it work Router wise?

Reply:
No, I mean if we remove the && !is_admin condition. We can't go to /username/xwallet even after we have already connected to our wallet. That's why I add that condition so we only redirected to /username only when we directly access the username/xwallet which is when the wallet haven't connected yet